samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb

```diff
@@ -200,19 +200,22 @@ def download_file(request)
         @tempfile = tempfile
       end
 
+      @stream = []
+
       # handle streaming Responses
       request.options.on_data = Proc.new do |chunk, overall_received_bytes|
         # puts "Received #{overall_received_bytes} characters"
-        tempfile.write(chunk)
+        @stream << chunk
```

Comment:
If you use a block, the file will be closed automatically.
Can this be used?
https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html#class-Net::HTTP-label-Streaming+Response+Bodies

Reply:
Could be. Previously I got IO error when using `tempfile.write(chunk)`.

Can you test your suggested fix locally and if it works, can you please file a PR?

Reply:
I see, I got it.